dags/elsevier/trigger_file_processing.py

```diff
@@ -0,0 +1,61 @@
+import base64
+import io
+import os
+
+from airflow.api.common import trigger_dag
+from common.pull_ftp import _generate_id
+from common.repository import IRepository
+from common.utils import (
+    parse_without_names_spaces,
+    process_archive_elsevier,
+    process_archive_specific_files,
+)
+from elsevier.metadata_parser import ElsevierMetadataParser
+from structlog import PrintLogger
+
+
+def trigger_file_processing_elsevier(
+    publisher: str,
+    repo: IRepository,
+    logger: PrintLogger,
+    filenames=None,
+    article_splitter_function=lambda x: [x],
+):
+    files = []
+    for filename in filenames:
+        logger.msg("Running processing.", filename=filename)
+        file_bytes = repo.get_by_id(filename)
+        (archive_file_content, s3_filename) = process_archive_elsevier(
+            file_bytes=file_bytes, file_name=filename
+        )
+        dataset_file = parse_without_names_spaces(archive_file_content.decode("utf-8"))
+        parser = ElsevierMetadataParser(file_path=filename)
+        parsed_articles = parser.parse(dataset_file)
+        for parsed_article in parsed_articles:
+            extracted_files = process_archive_specific_files(
+                file_bytes=file_bytes,
+                file_name=s3_filename,
+                specific_file=parsed_article["files"]["xml"],
+            )
+            full_file_path = list(extracted_files.keys())[0]
+            repo.save(full_file_path, io.BytesIO(extracted_files[full_file_path]))
+            logger.msg("Running processing.", filename=parsed_article["files"]["xml"])
+
+            file_bytes_extracted = repo.get_by_id(
+                os.path.join(*["extracted", parsed_article["files"]["xml"]])
+            )
+            for article in article_splitter_function(file_bytes_extracted):
+                _id = _generate_id(publisher)
+                encoded_article = base64.b64encode(article.getvalue()).decode()
+                trigger_dag.trigger_dag(
+                    dag_id=f"{publisher}_process_file",
+                    run_id=_id,
+                    conf={
+                        "file": encoded_article,
+                        "file_name": full_file_path,
+                        "metadata": parsed_article,
+                    },
+                    replace_microseconds=False,
+                )
+                files.append(full_file_path)
```

Comment:
Do we really need the three level for loop?

Reply:
You are right! Looking closer to it, I noticed we need 2 loops:
1. First loop parses the mapping files (dataset.xml) in order to get paths of xml files and journal info. 
https://github.com/cern-sis/workflows/blob/fb37938761c96d96f61d97739b5f073dcf128503/dags/elsevier/metadata_parser.py#L18-L51
In one dataset.xml file, there are more than one record mapping
2. Second loop iterates through the record files and parses them